pkg/xen/arch/x86_64/0003-Always-map-EfiRuntimeServicesCode-and-EfiRuntimeServ.patch

```diff
@@ -0,0 +1,33 @@
+From 0bda255f3085b83899a873393981dd284b8fc817 Mon Sep 17 00:00:00 2001
+From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
+Date: Sun, 23 Aug 2020 17:38:47 +0300
+Subject: [PATCH] Always map EfiRuntimeServicesCode and EfiRuntimeServicesData
+
+Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
+---
+ common/efi/boot.c | 8 +++++---
+ 1 file changed, 5 insertions(+), 3 deletions(-)
+
+diff --git a/common/efi/boot.c b/common/efi/boot.c
+index 5a520bf21d..14ef2c479a 100644
+--- a/common/efi/boot.c
++++ b/common/efi/boot.c
+@@ -1521,10 +1521,12 @@ void __init efi_init_memory(void)
+         }
+ 
+         if ( !efi_enabled(EFI_RS) ||
+-             (!(desc->Attribute & EFI_MEMORY_RUNTIME) &&
++             ((!(desc->Attribute & EFI_MEMORY_RUNTIME) &&
++                (desc->Type != EfiRuntimeServicesCode) &&
++                (desc->Type != EfiRuntimeServicesData)) &&
+               (!map_bs ||
+-               (desc->Type != EfiBootServicesCode &&
+-                desc->Type != EfiBootServicesData))) )
++               ((desc->Type != EfiBootServicesCode) &&
```

Comment:
This looks superfluous

Reply:
Aside from that, the patch looks useful if I guessed the intention correctly. It would be great if you could send it to xen-devel with a longer explanation of why it is needed and the bug that it fixes.

Thanks!

Reply:
It is basically a workaround for buggy UEFI implementations which do not conform to the spec and do not set the "Attributes" fields in the memory descriptors properly. It's the Dell IoT box FW which needs fixing - there can be other options, like a small memory map sanitizer DXE module. It's highly doubtful that it will be accepted in the upstream.

Reply:
I see. Keep in mind that the unfortunate reality is that upstream projects like Xen and Linux had to deal with buggy firmware for years, so we do have quirks and workarounds upstream to deal with some of them.

I don't know if this specific one will be one of them, but I think it would be good to send it to the list with this explanation. If nothing else it will be in the xen-devel archives and searchable.